src/steps/resource-manager/policy/client.ts

```
@@ -27,4 +31,103 @@ export class AzurePolicyClient extends Client {
       callback,
     });
   }
+
+  public async getPolicySetDefinition(
+    policySetDefinitionId: string,
+  ): Promise<PolicySetDefinition | undefined> {
+    const serviceClient = await this.getAuthenticatedServiceClient(
+      PolicyClient,
+    );
+
+    const name = getNameFromPolicyDefinitionId(policySetDefinitionId);
+    try {
+      const response = await serviceClient.policySetDefinitions.get(name);
+      return response;
+    } catch (err) {
+      this.logger.warn(
```

Should we throw an error here instead?

---

Yes I think that's a good idea.